src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotMetadataStep.java

```diff
@@ -45,14 +46,23 @@ public CreateSnapshotMetadataStep(
   public StepResult doStep(FlightContext context) {
     try {
       FlightMap workingMap = context.getWorkingMap();
-      // fill in the ideas that we made in previous steps
+      // fill in the ids that we made in previous steps
       UUID projectResourceId =
           workingMap.get(SnapshotWorkingMapKeys.PROJECT_RESOURCE_ID, UUID.class);
       Snapshot snapshot =
           snapshotService
               .makeSnapshotFromSnapshotRequest(snapshotReq)
               .id(snapshotId)
               .projectResourceId(projectResourceId);
+      if (snapshotReq
+          .getContents()
+          .get(0)
+          .getMode()
+          .equals(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID)) {
+        workingMap.put(
+            SnapshotWorkingMapKeys.SNAPSHOT_ASSET_SPECIFICATION,
+            snapshot.getFirstSnapshotSource().getAssetSpecification());
```

Comment:
I think trying to make the by request path match up with the regular path makes this part more confusing; the code generates an asset spec and adds it to the snapshot source, but it doesn't store it in the snapshot in the database. Do we need to generate the assset spec here at all, if the mode is `by request`?